### ISSUE TITLE
--remote-expr escapes special symbols for some reason

### ISSUE THREAD

**Issue by GoldsteinE** [neovim/neovim#18531]
### Neovim version (nvim -v)

0.8.0 commit 5359be78935dc639c481d74f010fe133dd40290c

### Vim (not Nvim) behaves the same?

no, Vim 8.2.2567

### Operating system/version

NixOS

### Terminal name/version

alacritty

### $TERM environment variable

xterm-256color

### Installation

https://github.com/nix-community/neovim-nightly-overlay/

### How to reproduce the issue

```
$ nvim --clean --listen /tmp/sock
# in the other terminal
$ nvim --clean --server "/tmp/sock" --remote-expr "$(printf '"\t\n"')"
^I
```

### Expected behavior

Client Neovim outputs a tab character followed by newline

### Actual behavior

Client Neovim outputs two characters `^I` verbatim followed by newline

**Comment by justinmk** [comment]
`--remote-expr` will likely be removed, see https://github.com/neovim/neovim/pull/18414 (read all discussion fully before commenting)

**Comment by hiberabyss** [comment]
Could use `nvr` instead. (`pip3 install --upgrade neovim-remote`)

**Comment by zeertzjq** [comment]
> It's also worth noting that exprs like `[1]` are printed in their Lua version (`table: 0x7f1f1ad747f8`). Vim outputs all array elems separated by newlines instead.

This can simply solved by surrounding the expression using `join()`.

**Comment by zeertzjq** [comment]
Fixed by #23980 #23988
